multiplayer/src/components/HostLobbyModal.tsx

```diff
@@ -0,0 +1,76 @@
+import { useContext, useState } from "react";
+import { Button } from "react-common/components/controls/Button";
+import { Input } from "react-common/components/controls/Input";
+import { Modal } from "react-common/components/controls/Modal";
+import { startGameAsync } from "../epics";
+import { clearModal } from "../state/actions";
+import { AppStateContext } from "../state/AppStateContext";
+
+export default function Render() {
+    const { state, dispatch } = useContext(AppStateContext);
+    const [ copySuccessful, setCopySuccessful ] = useState(false);
+    let inputRef: HTMLInputElement;
+
+    const onStartGameClick = async () => {
+        pxt.tickEvent("mp.hostlobby.startgame");
+        dispatch(clearModal());
+        await startGameAsync();
+    };
+
+    const handleCopyClick = () => {
+        if (pxt.BrowserUtils.isIpcRenderer()) {
```

Comment:
futureproofing. i like it. :)

Reply:
This was all shamelessly copied from https://github.com/microsoft/pxt/blob/multiplayer/thsparks/unify_game_window/react-common/components/share/ShareInfo.tsx#L64-L72 :)